mirakuru/base.py

```diff
@@ -18,16 +18,60 @@
 """Base executor with the most basic functionality."""
 
 import os
+import re
 import time
 import shlex
 import signal
 import subprocess
+import uuid
+import logging
 from contextlib import contextmanager
 
 from mirakuru.exceptions import TimeoutExpired, AlreadyRunning
 
 
-class Executor(object):
+log = logging.getLogger(__name__)
+
+
+class KidsMarker(object):
```

Comment:
I would be happy to split Executor even more as it got really big. E.g. timeout mechanisms could be abstracted out of Executor.